controllers/controlplaneupgrade_controller.go

```diff
@@ -363,3 +371,71 @@ func getCapiMachine(ctx context.Context, client client.Client, nodeUpgrade *anyw
 	}
 	return machine, nil
 }
+
+func cleanupKubeVipCM(ctx context.Context, log logr.Logger, client client.Client) error {
+	cm := &corev1.ConfigMap{}
+	if err := client.Get(ctx, GetNamespacedNameType(constants.KubeVipConfigMapName, constants.EksaSystemNamespace), cm); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("config map %s  not found, skipping deletion", "ConfigMap", constants.KubeVipConfigMapName, "Namespace", constants.EksaSystemNamespace)
+		} else {
+			return fmt.Errorf("getting %s config map: %v", constants.KubeVipConfigMapName, err)
+		}
+	} else {
+		log.Info("Deleting kube-vip config map", "ConfigMap", constants.KubeVipConfigMapName, "Namespace", constants.EksaSystemNamespace)
+		if err := client.Delete(ctx, cm); err != nil {
+			return fmt.Errorf("deleting %s config map: %v", constants.KubeVipConfigMapName, err)
+		}
+	}
+	return nil
+}
+
+func createKubeVipCMIfNotExist(ctx context.Context, client client.Client, cpUpgrade *anywherev1.ControlPlaneUpgrade) error {
+	kubeVipCM := &corev1.ConfigMap{}
+	if err := client.Get(ctx, GetNamespacedNameType(constants.KubeVipConfigMapName, constants.EksaSystemNamespace), kubeVipCM); err != nil {
+		if apierrors.IsNotFound(err) {
+			kubeVipCM, err = kubeVipConfigMap(cpUpgrade)
+			if err != nil {
+				return err
+			}
+			if err := client.Create(ctx, kubeVipCM); err != nil {
+				return fmt.Errorf("failed to create %s config map: %v", constants.KubeVipConfigMapName, err)
+			}
+		} else {
+			return fmt.Errorf("getting %s configmap: %v", constants.KubeVipConfigMapName, err)
+		}
+	}
+	return nil
+}
+
+func kubeVipConfigMap(cpUpgrade *anywherev1.ControlPlaneUpgrade) (*corev1.ConfigMap, error) {
+	kcpSpec, err := decodeAndUnmarshalKcpSpecData(cpUpgrade.Spec.ControlPlaneSpecData)
+	if err != nil {
+		return nil, err
+	}
+	var kubeVipConfig string
+	for _, file := range kcpSpec.KubeadmConfigSpec.Files {
+		if file.Path == kubeVipStaticPodPath {
+			kubeVipConfig = file.Content
+			break
+		}
+	}
```

Comment:
can you add a check here to see `kubeVipConfig` is not empty after the for loop? incase the kube-vip is not found in files, just error out